src/modeling.py

```diff
@@ -0,0 +1,228 @@
+    Args:
+      token_ids: <int32>[batch_size, long_seq_len] Tensor of token ids.
+      segment_ids: <int32>[batch_size, long_seq_len] Optional Tensor of segment
+        ids. By default we just fill all elements with segment id 1.
+      position_ids: <int32>[batch_size, long_seq_len] Optional Tensor of
+        absolute position ids. By default we use `tf.range(long_seq_len)` if
+        `max_absolute_position_embeddings` is nonzero. The maximum position id
+        must not be larger than `max_absolute_position_embeddings`.
+      att_mask: <int32>[batch_size, long_seq_len,  long_seq_len].
+      relative_att_ids: <int32>[batch_size, long_seq_len, long_seq_len].
+      long_embedding_adder: <float32>[batch_size, long_seq_len, hidden_size]
+        Tensor of additional values to add to the long input embedding before
+        layer normalization of the embeddings. By default this is skipped.
+      training: For Keras, optional boolean scalar tensor or Python boolean
+        indicating whether the call is meant for training or inference. Must
+        be None if `is_training` was not None in `__init__`.
+    Returns:
+      A list of Tensors, [long_output, global_output]:
+        long_output: <float32>[batch_size, long_seq_len, hidden_size]
+    """
+    if self.is_training is not None:
+      if training is not None:
+        raise ValueError(
+            "`training` must be None when `is_training` is given in `__init__`."
+        )
+      training = self.is_training
+
+    if segment_ids is None:
+      segment_ids = tf.ones_like(token_ids)
+
+    if self.config.max_absolute_position_embeddings == 0 and (
+        position_ids is not None):
+      raise ValueError(
+          "Cannot specify `position_ids` or `global_position_ids` arguments "
+          "when `max_absolute_position_embeddings` is 0.")
+
+    token_ids = token_ids.to_tensor()
+    long_input = self.token_embedding(token_ids)
+    long_input += self.segment_embedding(segment_ids)
+    if self.position_embedding is not None:
+      if position_ids is None:
+        long_input += self.position_embedding.embedding_table[
+            tf.newaxis, :long_input.shape[1], :]
+      else:
+        long_input += self.position_embedding(position_ids)
+    if long_embedding_adder is not None:
+      long_input += long_embedding_adder
+
+    if image_data is not None:
+
+      # Create padding to make image_data and long_input have the same shape.
+      batch_size, seq_len, hidden_size = tf_utils.get_shape_list(long_input)
+      _, image_seq_len, _ = tf_utils.get_shape_list(image_data)
+      prefix_pad = tf.zeros(
+          shape=(batch_size, 2, hidden_size),
+          dtype=tf.float32)
+      suffix_pad = tf.zeros(
+          shape=(batch_size, seq_len-2-image_seq_len, hidden_size),
+          dtype=tf.float32)
+      image_data = self.patch_embedding_transform(image_data)
+      
+      long_input += tf.concat([prefix_pad, image_data, suffix_pad], axis=1)
```

Comment:
Nit: It may be a bit more concise to use `tf.pad`.  https://www.tensorflow.org/api_docs/python/tf/pad